src/gui/toolbarMenubar/ToolPageSpinner.cpp

```diff
@@ -35,24 +62,52 @@ auto ToolPageSpinner::getNewToolIcon() -> GtkWidget* {
 }
 
 auto ToolPageSpinner::newItem() -> GtkToolItem* {
-    GtkToolItem* it = gtk_tool_item_new();
-
-    GtkWidget* hbox = gtk_box_new(GTK_ORIENTATION_HORIZONTAL, 1);
+    if (this->pageSpinner->hasWidget()) {
+        this->pageSpinner->removeWidget();
+    }
+    GtkWidget* spinner = gtk_spin_button_new_with_range(0, 1, 1);
+    this->pageSpinner->setWidget(spinner);
+    gtk_orientable_set_orientation(reinterpret_cast<GtkOrientable*>(spinner), orientation);
 
     GtkWidget* pageLabel = gtk_label_new(_("Page"));
-    GtkWidget* spinner = this->pageSpinner->getWidget();
+    this->lbPageNo = gtk_label_new("");
 
-    gtk_widget_set_valign(pageLabel, GTK_ALIGN_BASELINE);
-    gtk_box_pack_start(GTK_BOX(hbox), pageLabel, false, false, 7);
+    if (orientation == GTK_ORIENTATION_HORIZONTAL) {
+        this->lbVerticalPdfPage = nullptr;
+        gtk_widget_set_valign(pageLabel, GTK_ALIGN_BASELINE);
+        gtk_widget_set_valign(spinner, GTK_ALIGN_BASELINE);
+        gtk_widget_set_valign(this->lbPageNo, GTK_ALIGN_BASELINE);
+    } else {
+        this->lbVerticalPdfPage = gtk_label_new("");
+        gtk_widget_set_halign(pageLabel, GTK_ALIGN_BASELINE);
+        gtk_widget_set_halign(spinner, GTK_ALIGN_CENTER);
+        gtk_widget_set_halign(this->lbPageNo, GTK_ALIGN_BASELINE);
+        gtk_widget_set_halign(lbVerticalPdfPage, GTK_ALIGN_BASELINE);
+    }
 
-    gtk_widget_set_valign(spinner, GTK_ALIGN_BASELINE);
-    gtk_box_pack_start(GTK_BOX(hbox), spinner, false, false, 0);
+    updateLabels();
 
-    this->lbPageNo = gtk_label_new("");
-    gtk_widget_set_valign(this->lbPageNo, GTK_ALIGN_BASELINE);
-    gtk_box_pack_start(GTK_BOX(hbox), this->lbPageNo, false, false, 7);
+    box = gtk_box_new(orientation, 1);
```

Comment:
``updateLabels();`` line and ``box = gtk_box_new(orientation, 1);`` line should be in reverse order. Otherwise, if box is null, ``updateLabels()`` causes crash at ``gtk_box_pack_start`` in it. In addition, these two lines should be moved before the ``if-else clause`` just above them. Otherwise, parent widget checks in ``updateLabels()`` fails, and PDF page number won't shown.